src/main/java/org/opentripplanner/middleware/controllers/api/MonitoredTripController.java

```diff
@@ -158,22 +161,18 @@ private void verifyBelowMaxNumTrips(String userId, Request request) {
 
     /**
      * Checks that the given {@link MonitoredTrip} can be monitored (i.e., that the underlying
-     * {@link org.opentripplanner.middleware.otp.response.Itinerary} has transit and no rentals/ride hailing).
+     * {@link org.opentripplanner.middleware.otp.response.Itinerary} can be monitored).
      */
     private void checkTripCanBeMonitored(MonitoredTrip trip, Request request) {
-        boolean hasTransit = trip.itinerary.hasTransit();
-        boolean hasRentalOrRideHail = trip.itinerary.hasRentalOrRideHail();
-
-        if (!hasTransit || hasRentalOrRideHail) {
-            String rejectReason = "";
-            if (!hasTransit) rejectReason += "it does not include a transit leg";
-            if (!hasTransit && hasRentalOrRideHail) rejectReason += ", and ";
-            if (hasRentalOrRideHail) rejectReason += "it includes a rental or ride hail";
-
+        Set<InvalidItineraryReason> invalidReasons = trip.itinerary.checkItineraryIsMonitorable();
+        if (invalidReasons.size() > 0) {
```

Comment:
Could be replaced with !invalidReasons.isEmpty()

Reply:
Replaced in https://github.com/ibi-group/otp-middleware/pull/109/commits/c08f561d0b78149a3203db4dc9ae25744dfa36bc.